### PR TITLE
Added stop-gap APIs for agent release and install management

### PR DESCRIPTION
**Pull request by itzg**
# Resolves

Part of stop-gap solution for https://jira.rax.io/browse/SALUS-302

# What

Since we're too late in our "July 1 release" to introduce a new microservice, this PR is part of a stop-gap solution to introduce just enough api-admin direct usage of etcd to manage agent releases and installs in the currently supported approach with Ambassador.

# How

Adds back the api-admin's direct access to etcd and leverages the the `AgentCatalogService` that encapsulates the current logic the Ambassador assumes for looking up agent releases to push to the Envoys.

The added REST API is extremely minimal since this'll will very soon get ripped out for the proper implementation of the referenced Jira issue.

## How to test

All manually since this is a stop-gap

# Requires

The validation constraint changes from https://github.com/racker/salus-telemetry-model/pull/56

### CODE REVIEW

File: admin/src/main/java/com/rackspace/salus/telemetry/api/web/AgentInstallController.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2019 Rackspace US, Inc.
```
Comment (GeorgeJahad) [inline admin/src/main/java/com/rackspace/salus/telemetry/api/web/AgentInstallController.java]: is this class duplicating this one:
https://github.com/racker/salus-telemetry-api/blob/0fec9afd4310a1c599c963a450f63649b5230ffa/public/src/main/java/com/rackspace/salus/telemetry/api/web/AgentInstallationController.java

File: admin/src/main/java/com/rackspace/salus/telemetry/api/web/AgentReleasesController.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
+/*
```
Comment (GeorgeJahad) [inline admin/src/main/java/com/rackspace/salus/telemetry/api/web/AgentReleasesController.java]: looks like some overlap here as well:
https://github.com/racker/salus-telemetry-api/blob/0fec9afd4310a1c599c963a450f63649b5230ffa/public/src/main/java/com/rackspace/salus/telemetry/api/web/AgentReleaseController.java

is that intentional?

File: admin/src/main/java/com/rackspace/salus/telemetry/api/web/AgentInstallController.java (hunk 1 above)
Comment (itzg) [reply admin/src/main/java/com/rackspace/salus/telemetry/api/web/AgentInstallController.java]: Wow, good catch. I didn't even check to see if @jjbuchan had left the agent install controller over in public-api. I'll remove this one here in the admin, but fix those mapping paths to be kebab-case.

File: admin/src/main/java/com/rackspace/salus/telemetry/api/web/AgentReleasesController.java (hunk 2 above)
Comment (itzg) [reply admin/src/main/java/com/rackspace/salus/telemetry/api/web/AgentReleasesController.java]: This is intentional. The public api gets read-only access to see what agent releases can be used for the install operation. The controller methods here in the admin give us write-access to declare releases to be installed.

File: public/src/main/java/com/rackspace/salus/telemetry/api/web/AgentInstallationController.java [hunk 3]
```diff
@@ -38,7 +38,7 @@
  * personnel to manage deployment of agent releases for their customers.
  */
 @RestController
-@RequestMapping("/api/agentInstalls")
+@RequestMapping("/api/agent-installs")
```
Comment (GeorgeJahad) [inline public/src/main/java/com/rackspace/salus/telemetry/api/web/AgentInstallationController.java]: I'll update the insomnia workspace for these changes once this gets merged into master.

Comment (itzg) [reply public/src/main/java/com/rackspace/salus/telemetry/api/web/AgentInstallationController.java]: Thanks @GeorgeJahad

**Pull request merged**